airbyte-integrations/connectors/source-recharge/source_recharge/api.py

```diff
@@ -0,0 +1,179 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional
+
+import pendulum
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class RechargeStream(HttpStream, ABC):
+
+    primary_key = "id"
+    url_base = "https://api.rechargeapps.com/"
+
+    limit = 250
+    page_num = 1
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return self.name
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        stream_data = self.get_stream_data(response.json())
+        if len(stream_data) == self.limit:
+            self.page_num += 1
+            return {"page": self.page_num}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = {"limit": self.limit}
+        if next_page_token:
+            params.update(next_page_token)
+        if stream_slice:
+            params.update(stream_slice)
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_data = response.json()
+        stream_data = self.get_stream_data(response_data)
+
+        yield from stream_data
+
+    def get_stream_data(self, response_data: Any) -> List[dict]:
+        response_data = response_data.get(self.name, [])
+
+        if isinstance(response_data, list):
+            return response_data
+        elif isinstance(response_data, dict):
+            return [response_data]
+        else:
+            raise Exception(f"Unsupported type of response data for stream {self.name}")
+
+
+class IncrementalRechargeStream(RechargeStream, ABC):
+    def __init__(self, start_date, **kwargs):
+        super().__init__(**kwargs)
+        self._start_date = pendulum.parse(start_date)
+
+    cursor_field = "created_at"
```

Comment:
lets keep all class attributes on top of Class

Reply:
It is Not Incremental stream